packages/multichain/src/scope/assert.test.ts

```diff
@@ -0,0 +1,197 @@
+          'scopeString',
+          'eth_chainId',
+        );
+      });
+
+      it('throws an error if there are unsupported methods', () => {
+        MockSupported.isSupportedMethod.mockReturnValue(false);
+        expect(() => {
+          assertScopeSupported(
+            'scopeString',
+            {
+              ...validScopeObject,
+              methods: ['eth_chainId'],
+            },
+            {
+              isChainIdSupported,
+            },
+          );
+        }).toThrow(Caip25Errors.requestedMethodsNotSupportedError());
+      });
+
+      it('checks if the notifications are supported', () => {
+        MockSupported.isSupportedMethod.mockReturnValue(true);
+        try {
+          assertScopeSupported(
+            'scopeString',
+            {
+              ...validScopeObject,
+              notifications: ['chainChanged'],
+            },
+            {
+              isChainIdSupported,
+            },
+          );
+        } catch (err) {
+          // noop
+        }
+
+        expect(MockSupported.isSupportedNotification).toHaveBeenCalledWith(
+          'scopeString',
+          'chainChanged',
+        );
+      });
+
+      it('throws an error if there are unsupported notifications', () => {
+        MockSupported.isSupportedMethod.mockReturnValue(true);
+        MockSupported.isSupportedNotification.mockReturnValue(false);
+        expect(() => {
+          assertScopeSupported(
+            'scopeString',
+            {
+              ...validScopeObject,
+              notifications: ['chainChanged'],
+            },
+            {
+              isChainIdSupported,
+            },
+          );
+        }).toThrow(Caip25Errors.requestedNotificationsNotSupportedError());
+      });
+
+      it('does not throw if the scopeObject is valid', () => {
+        MockSupported.isSupportedMethod.mockReturnValue(true);
+        MockSupported.isSupportedNotification.mockReturnValue(true);
+        expect(
+          assertScopeSupported(
+            'scopeString',
+            {
+              ...validScopeObject,
+              methods: ['eth_chainId'],
+              notifications: ['chainChanged'],
+              accounts: ['eip155:1:0xdeadbeef'],
+            },
+            {
+              isChainIdSupported,
+            },
+          ),
+        ).toBeUndefined();
+      });
+    });
+  });
+
+  describe('assertScopesSupported', () => {
+    const isChainIdSupported = jest.fn();
+
+    it('does not throw an error if no scopes are defined', () => {
+      expect(
+        assertScopesSupported(
+          {},
+          {
+            isChainIdSupported,
+          },
+        ),
+      ).toBeUndefined();
+    });
+
+    it('throws an error if any scope is invalid', () => {
+      MockSupported.isSupportedScopeString.mockReturnValue(false);
+
+      expect(() => {
+        assertScopesSupported(
+          {
+            'eip155:1': validScopeObject,
+          },
```

Comment:
Nit: This would be a more effective test if only one scope was invalid.

The description implies that this would throw if it was a mix of valid/invalid, but the test case is all invalid.

```suggestion
      MockSupported.isSupportedScopeString.mockImplementation(
        (scopeString) => scopeString === 'eip155:1',
      );

      expect(() => {
        assertScopesSupported(
          {
            'eip155:1': validScopeObject,
            'eip155:2': validScopeObject,
          },
```